### PR TITLE
improve build settings

### PR DESCRIPTION
**Pull request by kotarella1110**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,29 +1,32 @@
...
   parser: '@typescript-eslint/parser',
+  plugins: ['@typescript-eslint', 'react-hooks'],
   extends: [
-    'plugin:@typescript-eslint/recommended',
     'plugin:react/recommended',
+    'plugin:@typescript-eslint/recommended',
+    'prettier/@typescript-eslint',
+    'plugin:prettier/recommended',
   ],
+  parser: '@typescript-eslint/parser',
```
Comment (pmaier983) [inline .eslintrc.js:10]: Is this a duplicate of the parser on line 2? ```parser: '@typescript-eslint/parser',```

Comment (kotarella1110) [reply .eslintrc.js:10]: @pmaier983 Oops! I will fix. thanks.

**Comment by kotarella1110** [comment]
`sourceMappinngURL` is added to the build file because changed rollup setting to output sourcemap file.
You don't want to output sourcemap file?

スクリーンショット 2020-06-29 13 12 08

**Comment by bluebill1049** [comment]
> `sourceMappingURL` is added to the build file because changed rollup setting to output sourcemap file.
> You don't want to output sourcemap file?
> 
> スクリーンショット 2020-06-29 13 12 08

Good to know, definitely want source map ❤️

File: package.json [hunk 2]
```diff
@@ -14,16 +14,16 @@
     "dist"
   ],
   "scripts": {
-    "clean": "rm -rf dist",
+    "clean": "rimraf dist",
```
Comment (bluebill1049) [inline package.json:17]: minor question: why do we need `rimraf` as a dependency? because of PC?

File: package.json [hunk 3]
```diff
@@ -14,16 +14,16 @@
...
     "prebuild": "yarn clean",
-    "build": "yarn build:modern && yarn build:umd && yarn build:ie11 && yarn build:min",
+    "build": "npm run clean && yarn build:modern && yarn build:umd && yarn build:ie11 && yarn build:min",
```
Comment (bluebill1049) [inline package.json:19]: let's use all `npm` 👍

Comment (kotarella1110) [reply package.json:19]: OK! I will 👍

File: package.json (hunk 2 above)
Comment (kotarella1110) [reply package.json:17]: In windows, unix command cannot be used.
I think it should be able to build on various OS.

Comment (bluebill1049) [reply package.json:17]: sounds good to me 👍

File: .github/workflows/unitTest.yml [hunk 4]
```diff
@@ -8,7 +8,15 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
-      - name: Run install
-        run: npm install
-      - name: Run test
-        run: npm run lint && npm run test
+      - name: Install dependencies
```
Comment (bluebill1049) [inline .github/workflows/unitTest.yml:11]: awesome! thanks for the clean up.

**Comment by bluebill1049** [comment]
LGTM! This will be the final build for V6.

**Review by bluebill1049 — APPROVED** [review]

**Pull request merged**